src/Raven.Server/Documents/DocumentIdWorker.cs

```diff
@@ -35,16 +38,15 @@ static DocumentIdWorker()
             var maxStrSize = Encoding.GetMaxByteCount(strLength);
             var escapePositionsSize = JsonParserState.FindEscapePositionsMaxSize(id);
 
-            
+            if (strLength > MaxIdSize)
```

Comment:
Isn't that too big? We are testing limitting this to 512 to have more breathing room

Reply:
The key issue, BTW, is that we use the document id in many places.
For example, a doc id that is max size + counter id that is max size, etc.

I think that we should have a hard limit of 512 and that is it.